kpops/utils/gen_schema.py

```diff
@@ -0,0 +1,137 @@
+import logging
+from enum import Enum
+from typing import Annotated, Any, Literal, Sequence, Union, get_args, get_origin
+
+from pydantic import Field, schema, schema_json_of
+from pydantic.fields import ModelField
+from pydantic.schema import SkipField
+from rich.console import Console
+
+from kpops.cli.pipeline_config import PipelineConfig
+from kpops.cli.registry import _find_classes
+from kpops.components.base_components.pipeline_component import PipelineComponent
+
+
+class SchemaScope(str, Enum):
+    pipeline = "pipeline"
+    config = "config"
+
+
+original_field_schema = schema.field_schema
+
+
+# adapted from https://github.com/tiangolo/fastapi/issues/1378#issuecomment-764966955
+def field_schema(field: ModelField, **kwargs: Any) -> Any:
+    if field.field_info.extra.get("hidden_from_schema"):
+        raise SkipField(f"{field.name} field is being hidden")
+    else:
+        return original_field_schema(field, **kwargs)
+
+
+schema.field_schema = field_schema
+
+log = logging.getLogger("")
+
+ModelField
+
+
+def is_valid_component(
+    used_fields: list[str], component: type[PipelineComponent]
+) -> bool:
+    """
+    Check whether a PipelineComponent subclass has a valid definition for the schema generation.
+
+    :param used_fields: types defined so far
+    :type used_fields: list[str]
+    :param component: component type to be validated
+    :type component: type[PipelineComponent]
+    :return: Whether component is valid for schema generation
+    :rtype: bool
+    """
+    component_name: str = component.__name__
+    # PipelineComponent always has a field 'type'
+    type: str = component.__fields__.get("type").field_info.default  # type: ignore[union-attr]
```

Comment:
`type` is a builtin function in Python, let's not override it. as a workaround you can change the variable name to something like `type_` or `typ`

Reply:
also consider using the classmethod which we've already added as a helper
`component.get_component_type()`

Reply:
done